src/reactviews/pages/QueryResult/table/plugins/cellSelectionModel.plugin.ts

```diff
@@ -419,6 +427,70 @@ export class CellSelectionModel<T extends Slick.SlickData>
         }
     }
 
+    private async setSelectionSummaryText(isSelection?: boolean) {
+        let summary = "";
+
+        if (isSelection) {
+            const selectedRanges: Slick.Range[] = this.getSelectedRanges();
+            const firstRange = selectedRanges[0];
+            if (!firstRange) return;
+
+            const column = this.grid.getColumns()[firstRange.fromCell];
+            if (!column) return;
+
+            const values: any[] = [];
+            let isNumeric = false;
+            let nullCount = 0;
+            let sum = 0;
+            let min = Infinity;
+            let max = -Infinity;
+
+            for (let row = firstRange.fromRow; row <= firstRange.toRow; row++) {
+                for (let col = firstRange.fromCell; col <= firstRange.toCell; col++) {
+                    const cell = this.grid.getCellNode(row, col);
+                    if (!cell) continue;
+                    const value = cell.innerText;
+                    const numValue = Number(value);
+                    if (value === "NULL") {
+                        nullCount++;
+                    } else if (!isNaN(numValue)) {
+                        isNumeric = true;
+                        min = Math.min(min, numValue);
+                        max = Math.max(max, numValue);
+                        sum += numValue;
+                    }
+                    values.push(value);
+                }
+            }
+
+            const count = values.length;
+            const distinctCount = new Set(values).size;
+
+            if (isNumeric) {
+                // format average into decimal, up to three places, with no trailing zeros
+                const average = (sum / count).toFixed(3).replace(/\.?0+$/, "");
+                summary = locConstants.queryResult.numericSelectionSummary(
+                    average,
+                    count,
+                    distinctCount,
+                    max,
+                    min,
+                    nullCount,
+                    sum,
+                );
+            } else {
+                summary = locConstants.queryResult.nonNumericSelectionSummary(
+                    count,
+                    distinctCount,
+                    nullCount,
+                );
+            }
+        }
+        await this.webViewState.extensionRpc.call("setSelectionSummary", {
```

Comment:
Why is this formatting this as a string, just to parse the pieces back out again in order to reformat them?  Seems like changing `summary` to be an interface with each stat as a prop would be cleaner.